lib/cloud_controller/deployment_updater/actions/scale.rb

```diff
@@ -0,0 +1,77 @@
+require 'cloud_controller/deployment_updater/actions/scale_down_canceled'
+require 'cloud_controller/deployment_updater/actions/scale_down_old_process'
+require 'cloud_controller/deployment_updater/actions/finalize'
+
+module VCAP::CloudController
+  module DeploymentUpdater
+    module Actions
+      class Scale
+        attr_reader :deployment, :logger, :app
+
+        def initialize(deployment, logger)
+          @deployment = deployment
+          @logger = logger
+          @app = deployment.app
+        end
+
+        def call
+          deployment.db.transaction do
+            return unless deployment.lock!.state == DeploymentModel::DEPLOYING_STATE
+            return unless all_instances_routable?
+
+            app.lock!
+            oldest_web_process_with_instances.lock!
+            deploying_web_process.lock!
+
+            deployment.update(
+              last_healthy_at: Time.now,
+              state: DeploymentModel::DEPLOYING_STATE,
+              status_value: DeploymentModel::ACTIVE_STATUS_VALUE,
+              status_reason: DeploymentModel::DEPLOYING_STATUS_REASON
+            )
+
+            if deploying_web_process.instances >= deployment.original_web_process_instance_count
+              Finalize.new(deployment).call
+              return
+            end
+
+            ScaleDownCanceled.new(deployment).call
```

Comment:
**meganit** I initially read this as "the scaling down is to be cancelled" rather than "scale down the canceled" process. Is there maybe a more expressive name? Also do we care at this level that they are cancelled? Could this be `ScaleDownIntermProcesses` or something?

Reply:
Well, it's not scaling down the Interim processes, only the Superseded and Canceled Processes.

I don't think this is a real case, btw--I can't get it to occur in my bosh-lite. Not completely sure, though

Reply:
Nevermind.  It's if you quickly deploy over a canceling deployment: https://github.com/cloudfoundry/cloud_controller_ng/blob/deployment-updater-refactor/spec/unit/actions/deployment_create_spec.rb#L451-L460